From 813019047d4c02b317f93107baced017d03ad872 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Thu, 25 Aug 2022 09:40:47 +0200
Subject: [PATCH] build: Drop libsoup API version check for flatpak library

The `soupapiversion` variable had been added in [1], but it did not
land yet and possibly never will. The flatpak library can be built
with libcurl instead, which should be detectable somehow too. As there
is currently no way to know what libsoup version the flatpak library
had been built with, if any, drop the check in the build script.

Closes https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1877

[1] https://github.com/flatpak/flatpak/pull/4582
---
 meson.build | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/meson.build b/meson.build
index eab142201..1acf08468 100644
--- a/meson.build
+++ b/meson.build
@@ -229,14 +229,6 @@ endif
 if get_option('flatpak')
   flatpak = dependency('flatpak', version : '>= 1.9.1')
   ostree = dependency('ostree-1')
-
-  flatpak_soupapiversion = flatpak.get_pkgconfig_variable('soupapiversion')
-  if flatpak_soupapiversion == ''
-    flatpak_soupapiversion = '2.4' # Pre libsoup3 support
-  endif
-  if flatpak_soupapiversion != libsoupapiversion
-    error('flatpak was built against a different API of libsoup. @0@ instead of @1@.'.format(flatpak_soupapiversion, libsoupapiversion))
-  endif
 endif
 
 if get_option('malcontent')
-- 
GitLab

